client/state/ui/selectors.js

```diff
@@ -90,3 +91,39 @@ export function isPreviewShowing( state ) {
 export function getPreviewUrl( state ) {
 	return state.ui.currentPreviewUrl;
 }
+
+/**
+ * Returns the current layout focus area
+ *
+ * @param  {Object}  state Global state tree
+ * @return {?String}  The current layout focus area
+ */
+export function getCurrentLayoutFocus( state ) {
+	// Temporary while we transition from the Flux store
+	return layoutFocusModule.getCurrent();
```

Comment:
After reviewing both versions of that change (#5208 and #6669), I'm still stuck with the same problem. Both of those cause the legacy action to be called when the redux store updates, but the problem here is that we want to update the redux store when the legacy action is called.

The only thing I can think of is to save a reference to the Redux store at some point just after it is created, then use that reference's dispatcher in the legacy Flux actions. I'll give that a try, but saving a reference to the store feels wrong somehow.